community/io/src/main/java/org/neo4j/io/pagecache/impl/muninn/MuninnReadPageCursor.java

```diff
@@ -60,10 +62,24 @@ public boolean next() throws IOException
         }
         pin( nextPageId, false );
         currentPageId = nextPageId;
+        verifyContext();
         nextPageId++;
         return true;
     }
 
+    private void verifyContext()
+    {
+        if ( versionContext.lastClosedTransactionId() == Long.MAX_VALUE )
+        {
+            return;
+        }
+        if ( page.getLastModifiedTxId() > versionContext.lastClosedTransactionId() ||
+             pagedFile.getHighestEvictedTransactionId() > versionContext.lastClosedTransactionId() )
```

Comment:
Also, read `versionContext.lastClosedTransactionId()` once instead of once per condition?